pkg/federation/sdlmerge/remove_empty_object_type_definition.go

```diff
@@ -0,0 +1,32 @@
+package sdlmerge
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+func newRemoveEmptyObjectTypeDefinition() *removeEmptyObjectTypeDefinition {
+	return &removeEmptyObjectTypeDefinition{}
+}
+
+type removeEmptyObjectTypeDefinition struct{}
+
+func (r *removeEmptyObjectTypeDefinition) Register(walker *astvisitor.Walker) {
+	walker.RegisterLeaveDocumentVisitor(r)
+}
+
+func (r *removeEmptyObjectTypeDefinition) LeaveDocument(operation, _ *ast.Document) {
```

Comment:
is it operation or one of sdl documents?

Reply:
This could be refactored, the problem is, we've named the "first" argument operation.
However, if we walk a SDL document, operation is the SDL and definition is nil.

Reply:
I don't want to touch it at this stage however.